### DISCUSSION TITLE
Packaging options for GNU/Linux

### DISCUSSION THREAD

**Discussion by starofrainnight** [taskcoach/taskcoach#6] [Ideas]
I think flatpak is now the best distribution way for distribute an application on linux system

1. RPM & DEB or other linux platform distribution required different package method on different platform (So many jobs need to do and so many bugs need to fix, etc ...)
2. AppImage is a good way, but building an AppImage appliaction requires an old version linux for compatible (So we can't use some new features)
3. snapcraft just like flatpak, but it's a closed source store solution (I don't like this idea)

Now the flatpak could install on almost every linux distribution, and we just need to pack for each cpu arch only, then it will magically works under each linux distribution

**Comment by wolftune** [comment]
I've had issues with flatpak, but they are mostly with programs like Tenacity which need access to other parts of the system (Jack in that case) and Flatpak support is limited. For Task Coach, I don't think that will be a problem.

I think it makes sense to prioritize one **easiest** to achieve and maintain option (so if that's Flatpak, great). The goal is definitely to have a reliable way for people to get Task Coach at all. When that is successful, *then* support for other options can be considered.

I agree strongly with disliking how Snap store has proprietary backend and embraces inclusion of proprietary software. That said, snapcraft itself appears FLO still, it's just the store on their server that is proprietary. I did have the impression that flathub includes proprietary software though I haven't fully investigated. I don't think this affects Task Coach directly enough. I'm fine with FlatPak.

**Reply by starofrainnight** [reply]
I agree about the flatpak have it's limitations, something like system tools (console program that access system directory or resources) seems not the target of Flatpak (I think it will never support them). But it seems to be the best packaging framework for general application packaging on linux so far, other ways all have hard part of development or distribution.

Actually I don't have think so far just like you did, I worte it down when the idea just pop up, nothing more (just in case I forget).
And you have enhance so much like goals and reasons, you are really such a thinker. Thanks! What you talk about that give the meaning about the idea and guide my way :smiley:

**Reply by wolftune** [reply]
Thanks, I think for this discussion into the future, we can be open here to anyone's thoughts on the packaging topic, but aiming for Flatpak per #7 issue can be the initial focus. It seems adequate and worth doing without taking extra time to do further comparison and discussion. Let's get Task Coach to the point where people are using it again more and requesting more packaging!

**Reply by starofrainnight** [reply]
Right, so be it :smile: !

**Comment by realcarbonneau** [comment]
Packaging is now done with automated github actions and we can review future needs when they arrive.   For now, the main linux distros and windows are covered.
